### ISSUE TITLE
[C#] Using expression-bodied functions with generics generates wrong type constraints.

### ISSUE THREAD

**Issue by Tuntenfisch** [doxygen/doxygen#10324]
**Describe the bug**
When using [expression-bodied functions](https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/statements-expressions-operators/expression-bodied-members) with [generics](https://learn.microsoft.com/en-us/dotnet/csharp/fundamentals/types/generics), the doxygen-generated documentation is wrong.

**Example**

```csharp
class Example
{
    static void Main(string[] args) => ExpressionBodiedFunction<int>();
    public static void ExpressionBodiedFunction<T>() where T : struct => System.Console.WriteLine("Example");
}
```

**Expected behavior**

![image](https://github.com/doxygen/doxygen/assets/11965324/869d4f8d-f05a-4efe-a7be-dda5264eedb2)

**Actual behavior**

![image](https://github.com/doxygen/doxygen/assets/11965324/b49ddd6a-845b-46c5-9593-80135b5ea945)

**To Reproduce**

[Example.zip](https://github.com/doxygen/doxygen/files/12714050/Example.zip)

**Version**

-  `doxygen --version`: 1.9.5
-  `winver`:  
    ![image](https://github.com/doxygen/doxygen/assets/11965324/12db943f-1fa4-47ad-aa5d-f2bd6fbabcf7)

**Comment by albert-github** [comment]
I took the example as provide wit the 1.9.5 doxygen version (also on windows. also I get these results with the current 1.9.8 doxygen version) and got:
![image](https://github.com/doxygen/doxygen/assets/5223533/64fc850c-255f-414f-b170-19ad81bf4292)

This looks like the actual behavior as you said.
- how did you get the actual behavior?

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #10326

**Comment by CodeSmile-0000011110110111** [comment]
This is the closest to "expression body" issue I could find. It's probably related.

When source code is embedded (I believe the parameter is INLINE\_SOURCES) then expression body members have no source code in their docs.

The method body gets embedded when the expression body is changed to bracket body.

**Comment by albert-github** [comment]
@CodeSmile-0000011110110111 

What do you want to express with your comment (is unclear to me)?
- that the proposed pull request solves the issue (also an issue you might have had)?
- there is another problem, in that case: 
  -  Can you please attach a, small, self contained example (source+configuration file in a, compressed, tar or zip file) that allows us to reproduce the problem? Please don't add external links as they might not be persistent.
  - Please also specify the **full** doxygen version used (`doxygen -v`) and whether or not you used the proposed patch.

**Comment by CodeSmile-0000011110110111** [comment]
I will test if the pull request fixes my issue but I'll wait until the next binary update. I was hoping to hear that this would include that fix.

I can make a new issue if unrelated.

**Comment by CodeSmile-0000011110110111** [comment]
I use doxygen 1.9.8. If the pull request is already in those binaries (it said "Open" so I assumed it isn't merged yet) I'll open a new issue in that case.

**Comment by albert-github** [comment]
The pull request is indeed still open, but you can test it with the executable of the Github Actions where the pull request is integrated based on the code base of September 26, 2023 (see  https://github.com/doxygen/doxygen/actions/runs/6310864577?pr=10326) unless you of course use a platform different from the one for which an executable is provided.

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.10.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).

**Comment by CodeSmile-0000011110110111** [comment]
Works for the one expression bodied generic method that I have. Type constraint is Object (UnityEngine.Object) as expected.
